packages/fx-core/tests/core/utils.ts

```diff
@@ -404,3 +405,58 @@ export function MockProjectSettings(appName: string): ProjectSettings {
     } as AzureSolutionSettings,
   };
 }
+
+export function MockPreviousContext(): Json {
+  return {
+    solution: {
+      teamsAppTenantId: "tenantId",
+      localDebugTeamsAppId: "teamsAppId",
+    },
+    "fx-resource-aad-app-for-teams": {
+      local_clientId: "local_clientId",
+      local_clientSecret: "{{fx-resource-aad-app-for-teams.local_clientSecret}}",
+      local_objectId: "local_objectId",
+      local_oauth2PermissionScopeId: "local_oauth2PermissionScopeId",
+      local_tenantId: "local_tenantId",
+      local_applicationIdUris: "local_applicationIdUris",
+    },
+  };
+}
+
+export function MockPreviousUserData(): Record<string, string> {
+  return {
+    "fx-resource-aad-app-for-teams.local_clientSecret": "local_clientSecret",
+  };
+}
+
+export function MockLatestContext(): Json {
```

Comment:
How about MockLatestVersion2_3_0Context, to make it more clear

Reply:
Fixed. Thanks!